### PR TITLE
resolve .link files for edge at install

### PR DESCRIPTION
**Pull request by giunatale**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -43,6 +43,8 @@
 import warnings
 from datetime import datetime
 
+from pynq.utils import download_overlays
```
Comment (yunqu) [inline setup.py:46]: This line errors out with `RuntimeError: No Devices Found` because it imports `pynq` then `ps`.

Comment (giunatale) [reply setup.py:46]: Understood. However, we need this changes if we want .link files. The only thing I can see doable is to disable this error if no devices are detected. Thoughts?

**Comment by giunatale** [comment]
I updated the PR. Could you test it and see if you get the error again?
If you do, could you please share the stack trace?

**Comment by giunatale** [comment]
Ok, so after a few tests with builds I realized the issues does not arise from this code, but from the fact that we do all sorts of things we should not be doing at `import pynq` time.

This is the stack trace
```

+ python3 -m pip install dist/pynq-2.6.0.tar.gz --upgrade --no-deps

Processing ./dist/pynq-2.6.0.tar.gz

    ERROR: Command errored out with exit status 1:

     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-au7pmag_/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-au7pmag_/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-bcac5k7d

         cwd: /tmp/pip-req-build-au7pmag_/

    Complete output (21 lines):

    /tmp/pip-req-build-au7pmag_/pynq/pl_server/device.py:82: UserWarning: No devices found, is the XRT environment sourced?

      UserWarning)

    Traceback (most recent call last):

      File "<string>", line 1, in <module>

      File "/tmp/pip-req-build-au7pmag_/setup.py", line 47, in <module>

        from pynq.utils import download_overlays

      File "/tmp/pip-req-build-au7pmag_/pynq/__init__.py", line 33, in <module>

        from .pl import PL

      File "/tmp/pip-req-build-au7pmag_/pynq/pl.py", line 39, in <module>

        from .ps import CPU_ARCH_IS_SUPPORTED, CPU_ARCH, ZYNQ_ARCH, ZU_ARCH

      File "/tmp/pip-req-build-au7pmag_/pynq/ps.py", line 322, in <module>

        class _ClocksUltrascale(_ClocksMeta):

      File "/tmp/pip-req-build-au7pmag_/pynq/ps.py", line 382, in _ClocksUltrascale

        IOPLL_CTRL = Register(CRL_APB_ADDRESS + IOPLL_CTRL_OFFSET)

      File "/tmp/pip-req-build-au7pmag_/pynq/registers.py", line 115, in __init__

        array = MMIO(address, np.dtype(register_type).itemsize).array

      File "/tmp/pip-req-build-au7pmag_/pynq/mmio.py", line 82, in __init__

        device = Device.active_device

      File "/tmp/pip-req-build-au7pmag_/pynq/pl_server/device.py", line 95, in active_device

        raise RuntimeError("No Devices Found")

    RuntimeError: No Devices Found

    ----------------------------------------

ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

@PeterOgden can you comment?

**Comment by giunatale** [comment]
Checks are passing now, I will merge the PR.

**Pull request merged**